File: 03Operadores/OperadoresUnar.java
public class OperadoresUnar
{
  public static void main(String[] args)
  {
    int a= -1, b= 0, c=0; 
    boolean bool= true; //dato de verdadero y falso
    
    System.out.println("Operadores postfijos");
    System.out.println("a++ = " + (a++) );
    System.out.println("a = " + a );
    System.out.println("a-- = " + (a--) );
    System.out.println("a = " + a );
    System.out.println("\nOperadores prefijos");
    // el \n anterior es una secuencia de escape
    System.out.println("b = " + b );
    System.out.println("++b = " + (++b) );
    System.out.println("b = " + b );
    System.out.println("--b = " + (--b) );
    System.out.println("b = " + b );
    //Observa bien el siguiente enunciado
    c = a++ + ++b;
    System.out.println("\nc = " + c );
    
    System.out.println("\nOperador unario !");
    System.out.println("!bool = " + !bool);
  }
}

File: 04Oper_SecCont/OperadoresUnar.cpp
#include <iostream>
//using namespace std;

int main(){
    int a= -1, b= 0, c= 0;
    bool d = true;
    std::cout << "Operadores postfijos" << std::endl;
    std::cout << "a++ = " << (a++) << std::endl;
    std::cout << "a = " << a << std::endl;
    std::cout << "a-- = " << (a--) << std::endl;
    std::cout << "a = " << a << std::endl;
    std::cout << "\nOperadores prefijos" << std::endl;
    // el \n anterior es una secuencia de escape
    std::cout << "b = " << b << std::endl;
    std::cout << "++b = " << (++b) << std::endl;
    std::cout << "b = " << b << std::endl;
    std::cout << "--b = " << (--b) << std::endl;
    std::cout << "b = " << b << std::endl;
    //Observa bien el siguiente enunciado
    c = a++ + ++b;
    std::cout << "\nc = " << c << std::endl;
    
    std::cout << "\nOperador unario !" << std::endl;
    std::cout << !3 <<" == " << !d << std::endl;

    return 0;
}

